### PR TITLE
Azure: Fixed ability to pass SSH key to Linux VMs

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
This PR fixes errors I found while testing the ability to stand up instances in Azure using an SSH key instead of a password.

### What issues does this PR fix or reference?
N/A

### Previous Behavior
Exception thrown looking for a password that isn't present in my config.

### New Behavior
Successful instance creation using the latest Azure SDK

### Tests written?
No

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/cloud/clouds/azurearm.py [hunk 1]
```diff
@@ -852,7 +852,7 @@ def create_network_interface(call=None, kwargs=None):
                     ]
                     break
             except CloudError as exc:
-                log.error('There was a cloud error: %s', exc)
+                log.error('There was a cloud error: {0}'.format(exc))
```
Comment (gtmanfred) [inline salt/cloud/clouds/azurearm.py]: please change this back to the other format, the previous method is faster.

**Review by gtmanfred — CHANGES_REQUESTED** [review]

**Comment by gtmanfred** [comment]
Everything else looks good.

File: salt/cloud/clouds/azurearm.py (hunk 1 above)
Comment (nicholasmhughes) [reply salt/cloud/clouds/azurearm.py]: as you wish

**Review by gtmanfred — APPROVED** [review]

**Review by cmcmarrow — APPROVED** [review]

**Comment by cmcmarrow** [comment]
Thanks nicholasmhughes. Looks good to me!

**Review by Ch3LL — CHANGES_REQUESTED** [review]
can we get a regression test added here so we can ensure this doesn't occur in the future?

**Comment by nicholasmhughes** [comment]
I don't have the time to create tests right now. We could throw it out there to the world to see if someone wants to take on writing tests, but I'm not sure if you want to hold up this bug fix.

**Comment by Ch3LL** [comment]
How long has this been an issue? is this a recent regression?

**Comment by nicholasmhughes** [comment]
No idea. It's been a year since I've used this cloud driver, and it was in the develop branch at that time.

**Comment by Ch3LL** [comment]
Added azurearm tests here: https://github.com/saltstack/salt/pull/54172

And cherry-picked your changes here into 2019.2.1 here https://github.com/saltstack/salt/pull/54173 since the tests do not pass without your changes. thanks!

**Comment by nicholasmhughes** [comment]
cool. we're good to close this now, right @Ch3LL ?

**Comment by Ch3LL** [comment]
I was actually hoping to merge this into 2019.2 at the same time we merge my PR in 2019.2.1.

**Comment by nicholasmhughes** [comment]
gotcha. _backs away slowly_

**Review by Ch3LL — APPROVED** [review]

**Comment by nicholasmhughes** [comment]
Going through my open PRs... @Ch3LL , is this still scheduled for merge?

**Comment by nicholasmhughes** [comment]
@Ch3LL this master branch move has me confused... Since this code made it into `2019.2.1`, which (I believe) was used for `master`, then we don't need this PR anymore... correct?

**Comment by waynew** [comment]
@nicholasmhughes If this code was in 2019.2.1 then it should be in master (and you can always double check, of course)

If it's already in master, then yeah you can feel free to just close this PR. Thanks!

**Comment by nicholasmhughes** [comment]
:boom:

**Pull request closed without merging**